react/components/IframeContainer.tsx

```diff
@@ -0,0 +1,56 @@
+import type { ReactNode } from 'react'
+import React from 'react'
+import { ThemeProvider } from '@vtex/admin-ui'
+
+import { IframeLegacy } from './IframeLegacy'
+import { Iframe } from './Iframe'
+import { DELOREAN_REGISTRY } from '../util'
+
+const isLegacy = (slug: string) =>
+  String(slug).startsWith('Site/') ||
+  String(slug) === 'a' ||
+  String(slug).startsWith('site/')
+
+const isDeloreanAdmin = (slug: string) =>
+  DELOREAN_REGISTRY.some(path => slug.startsWith(path))
+
+export function IframeContainer(props: Props) {
+  const { params, withoutLegacy, customHeightGap } = props
+
+  const slug = params?.slug ?? ''
+
+  if (withoutLegacy) {
+    // Cover just IO apps, ignore legacy apps.
+    return <Iframe params={params} />
+  }
+
+  if (isLegacy(slug)) {
+    // IframeLegacy and isLegacy are covering Catalog and Legacy CMS admins
+    // those are the old portal9 admins served by portal in vtexcommercestable
+    return (
+      <ThemeProvider>
```

Comment:
Why is using `ThemeProvider` here and not in the other Iframes?

Reply:
The other iframes already have it (it renders on each application)